clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -694,19 +694,18 @@ private void ensureProperTransactionalState() {
         if (transactionManager.isFenced())
             throw new ProducerFencedException("The current producer has been fenced off by a another producer using the same transactional id.");
```

Comment:
Good point, technically the request APIs allow it, but it shouldn't happen through our producer. I guess we can leave it.